### PR TITLE
parse alternative Message-ID email header too

### PR DESCRIPTION
**Pull request by eviljeff**

### CODE REVIEW

File: src/olympia/activity/tasks.py [hunk 1]
```diff
@@ -13,14 +13,21 @@
 @write
 def process_email(message, **kwargs):
     """Parse emails and save activity log entry."""
-    msg_id = message.get('MessageId', None)
+    # Some emails (gmail, at least) come with Message-ID instead of MessageId.
+    msg_id = message.get('MessageId')
+    if not msg_id:
+        custom_headers = message.get('CustomHeaders', [])
```
Comment (EnTeQuAk) [inline src/olympia/activity/tasks.py]: where is `CustomHeaders` originally coming from again?

Comment (eviljeff) [inline src/olympia/activity/tasks.py]: the message parameter is `.get('Message')`[[1](https://github.com/eviljeff/olympia/blob/f2dafe975078d7d811e28749867a3ac5570b460d/src/olympia/activity/views.py#L116)] from a payload like https://www.socketlabs.com/api-reference/inbound-api/#json

File: src/olympia/activity/tests/test_tasks.py [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+    # don't try to process the same message twice
+    process_email(message)
+    assert _mock.call_count == 1
+
+
+@mock.patch('olympia.activity.tasks.add_email_to_activity_log_wrapper')
+def test_process_email_different_messageid(_mock):
+    # Test 'Message-ID' works too.
+    message = {'CustomHeaders': [
+        {'Name': 'Message-ID', 'Value': '<gmail_tastic>'}]}
```
Comment (EnTeQuAk) [inline src/olympia/activity/tests/test_tasks.py]: Can we verify that we actually parsed and stored the correct message-id?

File: src/olympia/activity/tasks.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/activity/tasks.py]: Ah, perfect. I didn't see that, thanks!

File: src/olympia/activity/tests/test_tasks.py (hunk 2 above)
Comment (eviljeff) [inline src/olympia/activity/tests/test_tasks.py]: we don't actually store the message-id anywhere - it's just used as a cache-key so we don't double-process the emails.  As long as some non-null value is parsed it's okay.

**Pull request merged**